homeassistant/components/airvisual/__init__.py

```diff
@@ -295,86 +327,11 @@ def update():
             self.update_from_latest_data()
             self.async_write_ha_state()
 
-        self.async_on_remove(
-            async_dispatcher_connect(self.hass, self._airvisual.topic_update, update)
-        )
+        self.async_on_remove(self.coordinator.async_add_listener(update))
 
         self.update_from_latest_data()
 
```

Comment:
By implementing the entity `async_update` method and awaiting `coordinator.async_request_refresh` we allow users to use the common entity update service to update state.

Reply:
https://github.com/home-assistant/core/pull/35203